app/models/event.rb

```diff
@@ -0,0 +1,47 @@
+class Event
+
+  attr_reader :name, :start_datetime, :end_datetime, :description, :location, :link, :thumbnail
+
+	def initialize(event_json)
```

Comment:
Receiving `event_json` is not very good in my book. I would prefer it to simply get the parameters in order and let `pull_calendar_data` manage the creation.

So `Event.new(event_json['summary'], event_json['start'], ...)`. This makes the class and constructor simpler and leaves more work for the other methods/classes implementing it.

If you feel that passing in a hash makes it easier and is vastly used, you can always create another static method `self.from_hash` that creates a new `Event` based on the hash (kinda how you're currently doing)